File: bimestre1/aula_04-16/03-impares.go
package main
import "fmt"

// Dado um número inteiro positivo n, imprimir os n primeiros naturais ímpares
func main() {
	var n int

	fmt.Print("Informe a quantidade de números naturais ímpares desejada: ")
	fmt.Scanf("%d", &n)

	for n = n*2; n > 0; n -- {
		if n % 2 != 0 {
			fmt.Printf("%d\n", n);
		}
	}
}

File: bimestre1/aula_05-07/03-impares.hs
-- dado um número inteiro positivo n, imprimir os n primeiros naturais ímpares
imprimir :: Int -> IO ()
imprimir n = aux n 1
    where
        aux 0 _ = return ()
        aux k x = do
            print x
            aux (k - 1) (x + 2)

main :: IO ()
main = imprimir 5
